packages/next/src/build/utils.ts

```diff
@@ -1393,6 +1440,44 @@ export async function buildAppStaticPaths({
   )
 }
 
+function resolveAppConfig(generateParams: GenerateParams): AppConfig {
```

Comment:
This function is only called in one place, consider just constructing the AppConfig while you are producing the GenerateParams. No reason to re-traverse each layer when you are already doing it